x-pack/platform/test/functional_execution_context/tests/log_correlation.ts

```diff
@@ -29,24 +29,28 @@ export default function ({ getService }: FtrProviderContext) {
   const log = getService('log');
 
   describe('Log Correlation', () => {
-    // TODO: see https://github.com/elastic/kibana/pull/243499
-    it.skip('Emits "trace.id" into the logs', async () => {
+    it('Emits "trace.id" into the logs', async () => {
       if (isPrWithLabel('"ci:collect-apm"')) {
         log.warning(`Skipping test as APM is enabled in FTR, which breaks this test`);
         return;
       }
 
-      const response1 = await supertest.get('/emit_log_with_trace_id');
-      expect(response1.status).to.be(200);
-
-      expect(response1.body.traceId).to.be.a('string');
+      const traceId1 = '4bf92f3577b34da6a3ce929d0e0e4736';
+      const traceparent1 = `00-${traceId1}-00f067aa0ba902b7-01`;
 
-      const response2 = await supertest.get('/emit_log_with_trace_id');
+      const response1 = await supertest
+        .get('/emit_log_with_trace_id')
+        .set('traceparent', traceparent1);
+      expect(response1.status).to.be(200);
+      expect(response1.body.traceId).to.be(traceId1);
 
+      const traceId2 = 'a1b2c3d4e5f6789012345678901234ab';
+      const traceparent2 = `00-${traceId2}-10f067aa0ba902b7-01`;
+      const response2 = await supertest
+        .get('/emit_log_with_trace_id')
+        .set('traceparent', traceparent2);
       expect(response2.status).to.be(200);
-      expect(response1.body.traceId).to.be.a('string');
-
-      expect(response2.body.traceId).not.to.be(response1.body.traceId);
```

Comment:
TBH, I like the current solution of testing 2 explicitly different traceparents. However, it leaves one scenario uncovered (which is what I think that we were trying here before): if no `traceparent` is received, we assign different trace IDs to each request.

I'd like to see that scenario tested. However, it looks like, in some cases, since the FTR runner now has a parent traceId for the test, it is always the same.

Maybe the solution is to specify `.unset('traceparent')` to make sure that there's no traceparent in the request?

If that doesn't work, maybe we can conditionally check this assertion: if the runner sees a trace.id, it won't check this inequality. WDYT?